test/e2e/instrumentation/monitoring/custom_metrics_autoscaling.go

```diff
@@ -0,0 +1,205 @@
+		testHPA(f, kubeClient)
+	})
+})
+
+func testHPA(f *framework.Framework, kubeClient clientset.Interface) {
+	projectId := framework.TestContext.CloudConfig.ProjectID
+
+	ctx := context.Background()
+	client, err := google.DefaultClient(ctx, gcm.CloudPlatformScope)
+
+	// Hack for running tests locally, needed to authenticate in Stackdriver
+	// If this is your use case, create application default credentials:
+	// $ gcloud auth application-default login
+	// and uncomment following lines:
+	/*
+		ts, err := google.DefaultTokenSource(oauth2.NoContext)
+		framework.Logf("Couldn't get application default credentials, %v", err)
+		if err != nil {
+			framework.Failf("Error accessing application default credentials, %v", err)
+		}
+		client := oauth2.NewClient(oauth2.NoContext, ts)
+	*/
+
+	gcmService, err := gcm.New(client)
+	if err != nil {
+		framework.Failf("Failed to create gcm service, %v", err)
+	}
+
+	framework.ExpectNoError(err)
+
+	// Set up a cluster: create a custom metric and set up k8s-sd adapter
+	err = createDescriptors(gcmService, projectId)
+	if err != nil {
+		framework.Failf("Failed to create metric descriptor: %v", err)
+	}
+	defer cleanupDescriptors(gcmService, projectId)
+
+	err = createAdapter()
+	if err != nil {
+		framework.Failf("Failed to set up: %v", err)
+	}
+	defer cleanupAdapter()
+
+	// Run application that exports the metric
+	err = createDeploymentsToScale(kubeClient)
+	if err != nil {
+		framework.Failf("Failed to create stackdriver-exporter pod: %v", err)
+	}
+	defer cleanupDeploymentsToScale(kubeClient)
+
+	// Autoscale the deployments
+	err = createPodsHPA(kubeClient)
+	if err != nil {
+		framework.Failf("Failed to create 'Pods' HPA: %v", err)
+	}
+	err = createObjectHPA(kubeClient)
+	if err != nil {
+		framework.Failf("Failed to create 'Objects' HPA: %v", err)
+	}
+
+	// Wait a for HPA to scale down targets
+	// TODO: wait for the deployments to reach the expected numbers of replicas instead of fixed
+	//       amount of time
+	time.Sleep(240 * time.Second)
+
+	// Verify that the deployments were scaled down to the minimum value
+	sdExporterDeployment, err := kubeClient.Extensions().Deployments("default").Get("stackdriver-exporter-deployment", metav1.GetOptions{})
+	if err != nil {
+		framework.Failf("Failed to retrieve info about 'stackdriver-exporter-deployment'")
+	}
+	if *sdExporterDeployment.Spec.Replicas != 1 {
+		framework.Failf("Unexpected number of replicas for 'stackdriver-exporter-deployment'. Expected 1, but received %v", *sdExporterDeployment.Spec.Replicas)
+	}
+	dummyDeployment, err := kubeClient.Extensions().Deployments("default").Get("dummy-deployment", metav1.GetOptions{})
+	if err != nil {
+		framework.Failf("Failed to retrieve info about 'dummy-deployment'")
+	}
+	if *sdExporterDeployment.Spec.Replicas != 1 {
+		framework.Failf("Unexpected number of replicas for 'dummy-deployment'. Expected 1, but received %v", *dummyDeployment.Spec.Replicas)
+	}
+
+	framework.ExpectNoError(err)
```

Comment:
It looks like you already checked this error before in if on line 120.

Reply:
Removed.